### PR TITLE
feat(codemod): add tag to box transform

### PR DESCRIPTION
**Pull request by matheusps**
#### What is the purpose of this pull request?
- Created a codemod to transform deprecated `tag` to `Box` with the features: 

##### Package imports

```jsx
import { tag } from '@vtex/admin-ui'

⬇️ ⬇️ ⬇️ ⬇️ ⬇️

import { Box } from '@vtex/admin-ui'
```

##### Relative imports

```jsx
import { tag } from '../path'

⬇️ ⬇️ ⬇️ ⬇️ ⬇️

import { Box } from '../path'
```

##### JSX elements

```jsx
<tag.div />
<tag.img />
<tag.section></tag.section>

⬇️ ⬇️ ⬇️ ⬇️ ⬇️

<Box />
<Box as="img" />
<Box as="section"></Box>
```

##### Preserve as

```jsx
Import { Image } from 'next/image'
import { tag } from '@vtex/admin-ui'

<tag.img as={Image} />

⬇️ ⬇️ ⬇️ ⬇️ ⬇️

Import { Image } from 'next/image'
import { Box } from '@vtex/admin-ui'

<Box as={Image} />
```

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

**Review by lucasaarcoverde — APPROVED** [review]

File: packages/admin-ui-codemod/src/__tests__/tag-to-box.test.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    'import { tag, Button, Table } from "@vtex/admin-ui"',
+    'import { Box, Button, Table } from "@vtex/admin-ui"',
+    'package multiple imports'
+  )
+
+  defineInlineTest(
+    transform,
+    {},
+    'import { Tag, Button, Table } from "@vtex/admin-ui"',
+    'import { Tag, Button, Table } from "@vtex/admin-ui"',
```
Comment (Lenderson) [inline packages/admin-ui-codemod/src/__tests__/tag-to-box.test.js:25]: shouldn't it be?
```suggestion
    'import { Box, Button, Table } from "@vtex/admin-ui"',
```

File: packages/admin-ui-codemod/src/__tests__/tag-to-box.test.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    transform,
+    {},
+    'import { tag, Button, Table } from "../"',
+    'import { Box, Button, Table } from "../"',
+    'relative multiple imports'
+  )
+})
+
+describe('tag.[el] to box', () => {
+  defineInlineTest(transform, {}, '<Tag />', '<Tag />', 'case sensitive')
```
Comment (Lenderson) [inline packages/admin-ui-codemod/src/__tests__/tag-to-box.test.js:47]: similar here
```suggestion
  defineInlineTest(transform, {}, '<Tag />', '<Box />', 'case sensitive')
```

**Review by Lenderson — APPROVED** [review]
LGTM, just a few considerations

File: packages/admin-ui-codemod/src/__tests__/tag-to-box.test.js (hunk 1 above)
Comment (matheusps) [reply packages/admin-ui-codemod/src/__tests__/tag-to-box.test.js:25]: Nope. This grants that we don't mess with the [`Tag` component](https://admin-ui.vercel.app/guidelines/components/tag)

File: packages/admin-ui-codemod/src/__tests__/tag-to-box.test.js (hunk 2 above)
Comment (matheusps) [reply packages/admin-ui-codemod/src/__tests__/tag-to-box.test.js:47]: Same

**Pull request merged**